client/lib/user/support-user-interop.js

```diff
@@ -18,27 +25,99 @@ import { getSupportUser, getSupportToken } from 'state/support/selectors';
  *
  * @param {Object}  reduxStore  The global redux store instance
  */
-export default function( reduxStore ) {
-	const user = new User();
+class SupportUser {
+	constructor() {
+		debug( 'Support User is enabled in this environment' );
 
-	// Called when the support user token was updated in wpcom
-	const onTokenChange = () => {
-		user.fetch();
-		userSettings.fetchSettings();
+		this.reduxStoreReady = new Promise( ( resolve ) => {
+			this.setReduxStore = ( reduxStore ) => resolve( reduxStore );
+		} );
 	}
 
-	// Called when an API call fails due to a token error
-	const onTokenError = ( error ) => {
-		reduxStore.dispatch( supportUserRestore( error ) );
+	/**
+	 * Reboot normally as the main user
+	 */
+	rebootNormally() {
+		debug( 'Rebooting Calypso normally' );
+
+		localforage
```

Comment:
Did you consider storing it in `state.application`? We'll be persisting that state soonish and it may be a better fit.